module/validation/receipt_validator_test.go

```diff
@@ -229,7 +230,7 @@ func (s *ReceiptValidationSuite) TestReceiptInvalidPreviousResult() {
 		mock.Anything,
 		mock.Anything).Return(true, nil).Maybe()
 
-	err := s.receiptValidator.Validate(one(receipt))
+	err := s.receiptValidator.Validate(receipt)
 	s.Require().Error(err, "should reject invalid previous result")
```

Comment:
⚠️
I am worried that this test might be erroring on a different failure path than the one you intended to test. I believe your test returns with `UnverifiableInputError`:
* by changing the `PreviousResult` https://github.com/onflow/flow-go/blob/ea403fa5af245ee5c9668bd4c6c18d4fae4b587a/module/validation/receipt_validator_test.go#L241 you are also changing the `PreviousResult.ID`
* Since you are not updating `valSubgrph.Result`, the receipt being rejected due to a missing previous result. 
 
Essentially, you are testing a different error condition than the one you intended. This highlights the importance of not just expecting an error, but also verifying that the error is of the expected type:
```suggestion
	s.Require().Error(err, "should reject invalid previous result")
	s.Assert().True(engine.IsInvalidInputError(err), err)
```

Reply:
I've messed with ordering when moving tests, thanks for noticing it. Definitely makes sense.